erigon-lib/state/domain_shared.go

```diff
@@ -1214,6 +1214,9 @@ func (sdc *SharedDomainsCommitmentContext) LatestCommitmentState() (blockNum, tx
 		return 0, 0, nil, err
 	}
 	if len(state) < 16 {
+		if sdc.sharedDomains.roTtx.Debug().TxNumsInFiles(kv.CommitmentDomain) > 0 {
+			return 0, 0, nil, fmt.Errorf("assert: special key `state` not found in commitment files (but we have them), probably files are `purified`")
```

Comment:
after purification `state` key must persist as of latest value since purified are just merged kv's into one file

Reply:
or it supposed to be in db? anyways i assume it should be in files

Reply:
this case is not about "after purification" - it's about "after purification + manual run `erigon seg rm-state-snapshots --latest` couple times" - so only purified files left. Need failfast in this case we meaningful error.